src/map/lua/lua_baseentity.cpp

```diff
@@ -16848,6 +16848,79 @@ void CLuaBaseEntity::addTreasure(uint16 itemID, sol::object const& arg1, sol::ob
     }
 }
 
+/************************************************************************
+ *  Function: setStealItem()
+ *  Purpose : Used to reset the mob's steal table
+ *  Example : mob:setStealItem(xi.item.ITEM_NAME)
+ ************************************************************************/
+
+void CLuaBaseEntity::setStealItem(sol::object const& arg1)
+{
+    if (m_PBaseEntity->objtype != TYPE_MOB)
+    {
+        ShowWarning("Attempting to set steal item for invalid entity type (%s).", m_PBaseEntity->getName());
+        return;
+    }
+
+    // TODO support arg1 as a table
+    std::vector<uint16> Items;
+    if (arg1.is<uint16>())
+    {
+        auto item = arg1.as<uint16>();
+        if (itemutils::GetItemPointer(item))
+        {
+            Items.emplace_back(item);
+        }
+    }
+
+    if (Items.size() == 0)
+    {
+        ShowWarning("Attempting to set steal items with no valid itemids (%s).", m_PBaseEntity->getName());
+        return;
+    }
+
+    auto* PMob = dynamic_cast<CMobEntity*>(m_PBaseEntity);
+
+    if (PMob)
+    {
+        DropList_t* PDropList = itemutils::GetDropList(PMob->m_DropID);
```

Comment:
Is this modifying the _global droplist_ for this mob, which might be shared among other mobs?

I think that's the concern I have with this PR. I did a droplist modification PR at some point which handles this similarly by keeping track of a modification list per-mob, so the global entry isn't touched.

Reply:
It.... is

Definitely the wrong way to go here, oops. Got a link to that other PR?

Reply:
This was the original, just for funsies:
https://github.com/topaz-next/topaz/pull/2557

Reply:
Amazing. Looks like this listener is called when a mob dies and processes drops alongside the normal dropid? Should simplify the coppercap PR quite a lot.

Thanks. I'll just close this then and refactor the coppercap PR